### PR TITLE
Thread Web Agent

### PR DESCRIPTION
**Pull request by zahnno**
PR includes:

- Thread Web Agent.
- Provides a list of current active threads.
- Threads are clickable links that send params to browser which then redirects user to page with stack trace of the clicked thread.

### CODE REVIEW

File: src/foam/nanos/http/ThreadsWebAgent.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import java.io.PrintWriter;
+import javax.servlet.http.HttpServletRequest;
+
+public class ThreadsWebAgent
+  implements WebAgent
+{
+  public ThreadsWebAgent() {}
+
+  public void execute(X x) {
+    final PrintWriter out  = (PrintWriter) x.get(PrintWriter.class);
```
Comment (kgrgreer) [inline src/foam/nanos/http/ThreadsWebAgent.java:23]: Why is this final?

Comment (zahnno) [reply src/foam/nanos/http/ThreadsWebAgent.java:23]: PrintWriter out won't change.

File: src/foam/nanos/http/ThreadsWebAgent.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    HttpServletRequest req = (HttpServletRequest) x.get(HttpServletRequest.class);
+
+    Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
+
+    Thread[] threadArray = threadSet.toArray(new Thread[threadSet.size()]);
+    out.println("<HEAD><TITLE>Threads</TITLE></HEAD>\n");
+    out.println("<pre>");
+    out.println("<H1>Threads</H1>\n");
+
+    for (Thread thread : threadArray){
```
Comment (kgrgreer) [inline src/foam/nanos/http/ThreadsWebAgent.java]: Follow https://github.com/foam-framework/foam2/blob/master/doc/guides/StyleGuide.md for FOAM code.

File: src/foam/nanos/http/ThreadsWebAgent.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  public ThreadsWebAgent() {}
+
+  public void execute(X x) {
+    final PrintWriter out  = (PrintWriter) x.get(PrintWriter.class);
+    HttpServletRequest req = (HttpServletRequest) x.get(HttpServletRequest.class);
+
+    Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
+
+    Thread[] threadArray = threadSet.toArray(new Thread[threadSet.size()]);
+    out.println("<HEAD><TITLE>Threads</TITLE></HEAD>\n");
```
Comment (kgrgreer) [inline src/foam/nanos/http/ThreadsWebAgent.java]: Add BODY tags.

File: src/foam/nanos/http/ThreadsWebAgent.java (hunk 1 above)
Comment (kgrgreer) [reply src/foam/nanos/http/ThreadsWebAgent.java:23]: Then why isn't 'req' on the line below final also?

Comment (zahnno) [reply src/foam/nanos/http/ThreadsWebAgent.java:23]: true true

**Pull request merged**